### PR TITLE
Vitalizer 5

### LINKED ISSUE #1: Add support for 'browsers list' file with Babel ENV
**Reference:**
Will be supported when this PR is merged https://github.com/babel/babel-preset-env/pull/161

### LINKED ISSUE #11: Review Webpack import() functionality
**Reference**
- [ ] https://medium.com/faceyspacey/webpacks-import-will-soon-fetch-js-css-here-s-how-you-do-it-today-4eb5b4929852

### LINKED ISSUE #34: Add Code Splitting to Webpack
Moved from https://github.com/vital-software/dashboard/issues/4

- https://brotzky.co/blog/code-splitting-react-router-webpack-2/

### LINKED ISSUE #76: Rename config file
Rename the config file from `.build.config.js` to `.vitalizer.js`

### LINKED ISSUE #84: Update Readme
After v4 release we need to update the readme to reflect the current state of the stack.

Add a table of settings to env:

|     | DEV    | TEST | PROD  |
|---| -------|-------| ------|
|mode| 'development' | 'production' |'production' |
|HMR| ✅     | 🚫      |   🚫 |
|Minimizer| 🚫 | 🚫      |    ✅ |
|SourceMaps| inline | ext      |    ext |

### LINKED ISSUE #91: Remove pinned webpack-sources resolution when fix lands.
https://github.com/webpack/webpack/issues/5931

This patch is required in vitalizer and client projects:

```
"resolutions": {
    "webpack-sources": "1.0.1"
  }
```

### LINKED ISSUE #93: Upgrade from webpack-dev-server to webpack-serve

### LINKED ISSUE #102: Fix memory leak on development container
Crashes intermittently, terrible DX.

This is most likely caused by the HMR files accumulating in memory.
To observe this navigate to `localhost:3000/webpack-dev-server` and save the project to watch the files accumulate.

Some conversation around this issue here: https://github.com/webpack/webpack/issues/6929
May well be the way that we are using the webpack node API...

### PR DESCRIPTION
**Pull request by teriu**
## Purpose
Remove all non-webpack configuration and update webpack config/dependencies to use newer conventions.

Pretty sure this has finally fixed #102 too.

## Approach
- Remove global Babel config from this project (over in https://github.com/vital-software/babel-preset-vital-software)
- Remove global ESLint config from this project (over in https://github.com/vital-software/eslint-config)
- Remove global StyleLint config from this project (over in https://github.com/vital-software/stylelint-config-vital-software)
- Remove global Browserslist config (use project browserslist files instead)
- Remove custom webpack compiler setup (used for custom messages)
- Simplify and update CI pipeline tasks
- Simplify PostCSS config
- Switch to `webpack-serve` from `webpack-dev-server`
- Switch to using project `.env` file vs `.build.config.js` file
- Update to Node 10 from Node 8
- Use project `.browserslistrc` file
- Add Github PR template
- Add service worker support
- Add vendor chunking and polyfill module splitting support
- Add asset-manifest file to build
- Minify `index.html` file on build
- Use new `webpack-hot-client` for auto HMR support
- Use improved `[chunkhash:8]` for better asset cache support
- Fix LICENSE file
- Upgrade from Babel 6 to Babel 7
- Add `/healthcheck` url

Resolves #102 
Resolves #93
Resolves #91 
Resolves #84
Resolves #76
Resolves #34 
Resolves #11
Resolves #1

#### Open Questions and Pre-Merge TODOs
- [x] Review all dependencies for any that are stale
- [x] Update rules (under module)
- [x] Ensure commenting in webpack config and webpack-serve config is up-to-date
- [x] Update Readme #84

## Learning
- We can now use the [dynamic import()](https://developers.google.com/web/updates/2017/11/dynamic-import) feature to asynchronously load files. I'm using this in the updated Check In project to create chunks.

### CODE REVIEW

File: scripts/build.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    terminate the Node.js process with a non-zero exit code.
+ */
+process.on('unhandledRejection', (error) => {
+    throw error
+})
+
+const chalk = require('chalk')
+const fs = require('fs-extra')
+const webpack = require('webpack')
+// const bfj = require('bfj')
```
Comment (hally9k) [inline scripts/build.js]: Should we comment these commented out bits with why they are excluded and why they might be included in the future? Or delete them...

File: scripts/start.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+checkBrowsers(paths.appPath)
+    .then(() => printBrowsers(paths.appPath))
+    .then(() => serve(config(HOST, PORT)))
+    .catch((error) => {
+        if (error && error.message) {
+            console.log(error.message)
+        }
+
+        process.exit(1)
+    })
```
Comment (hally9k) [inline scripts/start.js]: This is nice and succinct! ✨

File: test/package.json [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+    "@babel/preset-flow": "^7.0.0-beta.51",
+    "@babel/preset-react": "^7.0.0-beta.51",
     "normalize.css": "^7.0.0",
     "react": "^16.2.0",
     "react-dom": "^16.2.0",
     "react-redux": "^5.0.7",
     "react-router": "^4.2.0",
     "react-router-dom": "^4.2.2",
-    "redux": "^3.7.2"
+    "redux": "^3.7.2",
```
Comment (hally9k) [inline test/package.json]: Should we upgrade these deps to match our projects? Maybe we could try and run greenkeeper on this subdirectory when the support for that is in too 🤔

File: test/perf/webpack.speed.bench.json [hunk 4]
```diff
@@ -1,103 +0,0 @@
-{
```
Comment (hally9k) [inline test/perf/webpack.speed.bench.json]: Are we still benching like this or did we scrap this test entirely?

File: test/task/e2e-simple.sh [hunk 5]
```diff
@@ -34,28 +34,10 @@ function handle_exit {
 # Check for the existence of one or more files.
 function exists {
   for f in $*; do
-    test -e "$f"
+    test -f "$f"
   done
 }
 
-# Check for a rise in the build speed
```
Comment (hally9k) [inline test/task/e2e-simple.sh]: Oh right so we just nuked this?

File: scripts/build.js (hunk 1 above)
Comment (teriu) [reply scripts/build.js]: Forgot about them, should be deleted!

File: test/package.json (hunk 3 above)
Comment (teriu) [reply test/package.json]: Doesn't really matter, it's more to ensure Vitalizer runs over code correctly.

File: test/perf/webpack.speed.bench.json (hunk 4 above)
Comment (teriu) [reply test/perf/webpack.speed.bench.json]: Scrapped it, we can look at adding back in down the line if it's still an issue.

File: test/task/e2e-simple.sh (hunk 5 above)
Comment (teriu) [reply test/task/e2e-simple.sh]: Yep, not a big concern for us at this stage. I'd prefer to spend time looking into individual project bundle sizes, and using dynamic imports to speed up compilation.

**Pull request merged**